shuup/core/utils/context_cache.py

```diff
@@ -99,21 +121,36 @@ def bump_cache_for_product(product, shop=None):
     In case shop is not given all the shop products
     for the product is bumped.
 
-    :param product: product object
+    :param product: product object or product object id or a list of product object id's
     :type product: shuup.core.models.Product
     :param shop: shop object
     :type shop: shuup.core.models.Shop|None
     """
     from shuup.core.models import ShopProduct
-    if not shop:
-        for sp in ShopProduct.objects.filter(product_id=product.id):
-            bump_cache_for_shop_product(sp)
+
+    if not isinstance(product, list):
+        product_id = (product.id if hasattr(product, "id") else product)
+        products = [product_id]
     else:
-        try:
-            shop_product = product.get_shop_instance(shop=shop, allow_cache=False)
-            bump_cache_for_shop_product(shop_product)
-        except ShopProduct.DoesNotExist:
-            bump_cache_for_item(product)
+        products = product
+
+    for sp_id in ShopProduct.objects.filter(product_id__in= products):
```

Comment:
- remove the extra space from the `= products`
- when you iterate over a queryset like this, we are wasting time fetching all ShopProduct data from database, loading and populating it to the object when we just need their ids. Can we do it like this:

```py
shop_product_ids = ShopProduct.objects.filter(product_id__in=products).values_list("pk", flat=True)
for shop_product_id in shop_product_ids:
    bump_cache_for_shop_product(shop_product_id)
```

Reply:
Yeah, I'll fix that. Not used to writing such performance intensive code so my brain always defaults to fetching objects.